tests/rados/test_add_db_wal_bst.py

```diff
@@ -0,0 +1,138 @@
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    ceph_nodes = kw.get("ceph_nodes")
+    rados_object = RadosOrchestrator(node=cephadm)
+    bluestore_obj = BluestoreToolWorkflows(node=cephadm)
+    osd_flag_status = 0
+    try:
+        # Get the OSD list
+        for node in ceph_nodes:
+            if node.role == "osd":
+                host_name = node.hostname
+                ssd_path_list = rados_object.get_available_ssd(host_name)
+                if not ssd_path_list:
+                    log.info(f"The device paths are empty to add on the {host_name}")
+                    continue
+                osds_id_in_node = rados_object.collect_osd_daemon_ids(node)
+                log.info(f"The osd's  configured on the HDD are-{osds_id_in_node}")
+                test_osd = random.choice(osds_id_in_node)
+                log.info(f"Performing the configurations on {test_osd}")
+
+                nvme_flag = False
+                ssd_flag = False
+                for dev_path in ssd_path_list:
+                    if "nvme" in dev_path and not nvme_flag:
+                        stop_osd(rados_object, test_osd)
+                        bluestore_obj.add_wal_device(test_osd, dev_path)
+                        start_osd(rados_object, test_osd)
+                        osd_config_result = check_osd_config(
+                            rados_object, host_name, dev_path
+                        )
+                        if not osd_config_result:
+                            log.error(
+                                f"The nvme path {dev_path} is not added to the osd"
+                            )
+                            return 1
+                        nvme_flag = True
+                    elif not ssd_flag:
+                        stop_osd(rados_object, test_osd)
+                        bluestore_obj.add_db_device(test_osd, dev_path)
+                        start_osd(rados_object, test_osd)
+                        osd_config_result = check_osd_config(
+                            rados_object, host_name, dev_path
+                        )
+                        if not osd_config_result:
+                            log.error(
+                                f"The SSD path {dev_path} is not added to the osd"
+                            )
+                            return 1
+                        ssd_flag = True
+
+                if nvme_flag:
+                    log.info(f"The NVMe is configured to the {test_osd}")
+                else:
+                    log.info(f"The NVMe is not configured to the {test_osd}")
+                if ssd_flag:
+                    log.info(f"The SSD is configured to the {test_osd}")
+                else:
+                    log.info(f"The SSD is not configured to the {test_osd}")
+                osd_flag_status = osd_flag_status + 1
+
+        if osd_flag_status == 0:
+            log.error("Any of the devices are not configured as OSDs")
```

Comment:
As we are not trying on all the devices/OSDs, it would be good to capture and log a final list of OSDs for which WAL/DB addition was tried.
Moreover, IMO, if the wal/db addition on any of device fails, the test execution may not even reach this code, wdyt?

Reply:
The Logic is if none of the devices are not available to add the OSD or if any one device(DB/WAL)  is added to the OSD then the test case will pass.